pkg/rancher/rancher.go

```diff
@@ -283,11 +289,38 @@ func New(ctx context.Context, clientConfg clientcmd.ClientConfig, opts *Options)
 		return nil, err
 	}
 
-	auditLogWriter := audit.NewLogWriter(opts.AuditLogPath, audit.Level(opts.AuditLevel), opts.AuditLogMaxage, opts.AuditLogMaxbackup, opts.AuditLogMaxsize)
-	auditFilter, err := audit.NewAuditLogMiddleware(auditLogWriter)
+	var auditLogOut io.Writer
+
+	if opts.AuditLogEnabled {
+		out := &lumberjack.Logger{
+			Filename:   opts.AuditLogPath,
+			MaxAge:     opts.AuditLogMaxage,
+			MaxBackups: opts.AuditLogMaxbackup,
+			MaxSize:    opts.AuditLogMaxsize,
+		}
+		defer out.Close()
```

Comment:
Looks like this `New()` function is not blocking, it returns after the controller's are registered. Is `defer out.Close()` what we want here?